src-schema/Migration.php

```diff
@@ -11,745 +11,209 @@
-    {
-        $this->mode('alter')->execute();
-
-        return $this;
-    }
-
-    /**
-     * Rename table.
-     *
-     * @return $this
-     */
-    public function rename(): self
-    {
-        $this->mode('rename')->execute();
-
-        return $this;
-    }
-
-    /**
-     * Will read current schema and consult current 'field' arguments, to see if they are matched.
-     * If table does not exist, will invoke ->create. If table does exist, then it will execute
-     * methods ->newField(), ->dropField() or ->alterField() as needed, then call ->alter().
-     *
-     * @return string Returns short textual info for logging purposes
-     */
-    public function run(): string
-    {
-        $changes = $added = $altered = $dropped = 0;
-
-        // We use this to read fields from SQL
-        $migration2 = new static($this->connection);
-
-        if (!$migration2->importTable($this['table'])) {
-            // should probably use custom exception class here
-            $this->create();
-
-            return 'created new table';
-        }
-
-        $old = $migration2->_getFields();
-        $new = $this->_getFields();
-
-        // add new fields or update existing ones
-        foreach ($new as $field => $options) {
-            // never update ID field (sadly hard-coded field name)
-            if ($field === 'id') {
-                continue;
-            }
-
-            if (isset($old[$field])) {
-                // compare options and if needed alter field
-                // @todo add more options here like 'len'
-                if (array_key_exists('type', $old[$field]) && array_key_exists('type', $options)) {
-                    $oldSQLFieldType = $this->getSqlFieldType($old[$field]['type']);
-                    $newSQLFieldType = $this->getSqlFieldType($options['type']);
-                    if ($oldSQLFieldType !== $newSQLFieldType) {
-                        $this->alterField($field, $options);
-                        ++$altered;
-                        ++$changes;
-                    }
-                }
-
-                unset($old[$field]);
-            } else {
-                // new field, so let's just add it
-                $this->newField($field, $options);
-                ++$added;
-                ++$changes;
-            }
-        }
-
-        // remaining old fields - drop them
-        foreach ($old as $field => $options) {
-            // never delete ID field (sadly hard-coded field name)
-            if ($field === 'id') {
-                continue;
-            }
-
-            $this->dropField($field);
-            ++$dropped;
-            ++$changes;
+        try {
+            $this->getSchemaManager()->dropTable($this->getDatabasePlatform()->quoteSingleIdentifier($this->table->getName()));
+        } catch (\Doctrine\DBAL\DBALException $e) {
+        }
+
+        if ($this->getDatabasePlatform() instanceof OraclePlatform) {
+            // drop trigger if exists
```

Comment:
Probably should first drop trigger and only then drop table.

Reply:
fixed